### PR TITLE
Implement canon_path_in_place for backslashes

### PR DESCRIPTION
**Pull request by clemenswasser**
I also changed the tests, so that they test for both path separators.
@evmar  FYI: This was the final blocker for successfully building CMake on Windows with CMake + n2 😄 
However I still need to find out, why it is re-running CMake every time, even if nothing has changed... Or is this normal?

### CODE REVIEW

**Comment by clemenswasser** [comment]
I also tried out building LLVM with CMake + n2 on Windows. It also re-runs CMake on every invocation (I'm now sure, that this is definitely a Bug) and currently fails on:
```
failed: Building ASM_MASM object lib\Support\BLAKE3\CMakeFiles\LLVMSupportBlake3.dir\blake3_avx2_x86-64_windows_msvc.asm.obj
```

**Comment by evmar** [comment]
Re re-running: https://github.com/evmar/n2/issues/40

My recollection is that Ninja on Windows had to do some really tricky stuff to properly model paths with both forward and back slashes.  I wonder if this ends up not sufficient.

File: src/canon.rs [hunk 1]
```diff
@@ -136,33 +136,38 @@ pub fn canon_path<T: Into<String>>(inpath: T) -> String {
...
+        assert_canon_path_eq(&[".", "foo"], &["foo"]);
+        assert_canon_path_eq(&["foo", "."], &["foo", ""]);
+        assert_canon_path_eq(&["foo", ".", "bar"], &["foo", "bar"]);
     }
 
     #[test]
     fn slash() {
-        assert_eq!(canon_path("/foo"), "/foo");
-        assert_eq!(canon_path("foo//bar"), "foo/bar");
+        assert_canon_path_eq(&["", "foo"], &["", "foo"]);
```
Comment (evmar) [inline src/canon.rs:161]: I think these would be better if assert_canon_path_eq accepted strings with "/" and internally did the / => \ replacement.

**Comment by evmar** [comment]
https://github.com/evmar/n2/issues/42 some design thoughts on how to do this

**Pull request merged**